src/components/ProblemsPage/Selection.tsx

```diff
@@ -1,26 +1,62 @@
-import * as React from 'react';
+import React, { useEffect, useState } from 'react';
+import {
+  UseRefinementListProps,
+  useInstantSearch,
+  useRefinementList,
+} from 'react-instantsearch';
 import Select from 'react-select';
 import { useDarkMode } from '../../context/DarkModeContext';
 
-export default function Difficulty({ items, refine }) {
-  const darkMode = useDarkMode();
-
-  const handleChange = e => {
-    const refinements = [];
-    for (const difficulty of e) {
-      refinements.push(difficulty.label);
-    }
-    refine(refinements);
-  };
+export type SelectionProps = UseRefinementListProps & {
+  placeholder: string;
+  searchable: boolean;
+  isMulti: boolean;
+  transformLabel?: (label: string) => string;
+  items?: { label: string; value: string | string[] }[];
+};
 
+export default function Selection({
+  attribute,
+  limit,
+  placeholder,
+  searchable,
+  isMulti,
+  transformLabel: transform,
+  items,
+  ...props
+}: SelectionProps) {
+  const { items: refineItems } = useRefinementList({
+    attribute,
+    limit,
+    ...props,
+  });
+  if (!items) items = refineItems;
+  const darkMode = useDarkMode();
+  const [refinements, setRefinements] = useState<string[]>([]);
+  const { setIndexUiState } = useInstantSearch();
+  useEffect(() => {
+    setIndexUiState(prevIndexUiState => ({
+      refinementList: {
+        ...prevIndexUiState.refinementList,
+        [attribute]: refinements,
+      },
+    }));
+  }, [refinements]);
```

Comment:
if `refinements` is only used in `useEffect`, it may make more sense to just define a `setRefinements` function that directly calls `setIndexUiState` without having the `useEffect` wrapper. It doesn't really matter though.

Reply:
Yeah I think we can just keep it as is for now 😅